app.js

```diff
@@ -18,4 +18,248 @@ links.forEach((link) => {
+        <li>css</li>
+        <li><img src="img/Devider.svg" alt="skills divider line"></li>
+      </ul>
+    </div>
+    <div class="popup-btn-container">
+      <button class="button btn-pop">
+        <a href="${project.liveVersionLink}">See Live</a>
+        <img src="img/pop-up-live.svg">
+      </button>
+      <button class="button btn-pop">
+        <a href="${project.sourceLink}">See Source</a>
+        <img src="img/pop-up-source.svg">
+      </button>
+    </div>
+  </div>
+  `;
+  const contentWrapper = document.createElement('div');
+  contentWrapper.classList.add('content-wrapper');
+  const nameElement = document.createElement('h2');
+  const imageElement = document.createElement('a');
+  const descriptionElement = document.createElement('p');
+  const technologiesElement = document.createElement('div');
+
+  contentWrapper.append(
+    popupDetails,
+    nameElement,
+    imageElement,
+    descriptionElement,
+    technologiesElement,
+  );
+  popup.append(closeContainer, contentWrapper);
+
+  closeContainer.appendChild(closeButton);
+
+  closeButton.addEventListener('click', () => {
+    popup.remove();
+  });
+
+  return popup;
+}
+
+projectSection.addEventListener('click', ({ target }) => {
+  if (target.tagName === 'BUTTON') {
+    const { projectIndex } = target.dataset;
+    const project = projects[projectIndex];
+    const popup = createProjectDetailsPopup(project);
+    document.body.appendChild(popup);
+  }
+});
+
+function generateProjectElements() {
+  projectSection.innerHTML = '';
+
+  for (let i = 0; i < projects.length; i += 1) {
+    const project = projects[i];
+    const projectElement = document.createElement('div');
+    projectElement.classList.add('project');
+    projectElement.innerHTML = `
+    <div class="img-place-container ${project.gradientClass}">
+      <img src="${project.featuredImage}" alt="${project.name}">
+      <div class="${project.gradientContainer}">
+        <img src="${project.gradientImg}" alt="background gradient">
+      </div>
+    </div>
+    <div class="heading">
+      <h3 class="work-header">${project.name}</h3>
+    </div>
+    <div class="description">
+      <p class="work-description">${project.description}</p>
+    </div>  
+      
+    <div class="work-skills-container">
+      <ul class="work-skills">
+        <li>${project.technologies[0]}</li>
+        <li><img src="img/Rectangle 70.svg" alt="skills divider line"></li>
+        <li>${project.technologies[1]}</li>
+        <li><img src="img/Rectangle 70.svg" alt="skills divider line"></li>
+        <li>${project.technologies[2]}</li>
+        <li><img src="img/Rectangle 70.svg" alt="skills divider line"></li>
+        <li>${project.technologies[3]}</li>
+      </ul>
+    </div>
+
+    <div class="button-container">
+      <button class='button' data-project-index="${i}">See Project</button>
+    </div>
+    `;
+    projectSection.appendChild(projectElement);
+  }
+}
+generateProjectElements();
```

Comment:
Wow 👏🏾 👏🏾 👏🏾 👏🏾 👏🏾 👏🏾 

Beautifully crafted. I tell you, in all projects, I have reviewed this is very neatly done. well done.

Reply:
Thank you sir, much appreciated.